### PR TITLE
Add identity bloq

### PR DESCRIPTION
**Pull request by fdmalone**
It's helpful to have this when (for example) building a PrepareOracle for ReflectionUsingPrepare to achieve a reflection about zero ($P=I$).

Part of #998

### CODE REVIEW

File: qualtran/bloqs/basic_gates/identity.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
+#  Copyright 2023 Google LLC
```
Comment (tanujkhattar) [inline qualtran/bloqs/basic_gates/identity.py:1]:
```suggestion
#  Copyright 2024 Google LLC
```

File: qualtran/bloqs/basic_gates/identity.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    import cirq
+    import quimb.tensor as qtn
+
+    from qualtran.cirq_interop import CirqQuregT
+    from qualtran.simulation.classical_sim import ClassicalValT
+
+
+@frozen
+class Identity(Bloq):
+    r"""The identity gate
```
Comment (tanujkhattar) [inline qualtran/bloqs/basic_gates/identity.py:44]:
```suggestion
    r"""The identity gate on 1 qubit
```

File: qualtran/bloqs/basic_gates/identity.py (hunk 1 above)
Comment (fdmalone) [reply qualtran/bloqs/basic_gates/identity.py:1]: Done.

File: qualtran/bloqs/basic_gates/identity.py (hunk 2 above)
Comment (fdmalone) [reply qualtran/bloqs/basic_gates/identity.py:44]: Done.

**Review by tanujkhattar — APPROVED** [review]

**Pull request merged**